libs/utils/src/http/endpoint.rs

```diff
@@ -63,8 +63,22 @@ async fn prometheus_metrics_handler(_req: Request<Body>) -> Result<Response<Body
     Ok(response)
 }
 
+pub fn add_request_id_middleware<B: hyper::body::HttpBody + Send + Sync + 'static>(
+) -> Middleware<B, ApiError> {
+    Middleware::pre(move |mut req| async move {
+        let headers = req.headers_mut();
+        let name = HeaderName::from_str("UUID").unwrap();
+        let request_id = uuid::Uuid::new_v4().to_string();
+        let value = HeaderValue::from_str(&request_id).unwrap();
+        headers.insert(name, value);
+        tracing::info!("{} {} {}", req.method(), req.uri().path(), request_id);
```

Comment:
Should it be debug or something else to avoid increase in log volume?

See https://github.com/neondatabase/neon/pull/3681